### PR TITLE
[ENH] full support for `Panel.numpyflat` mtype in `datatypes` module, checks/conversions

### PR DESCRIPTION
**Pull request by fkiraly**
This PR adds full checking and conversion support for the `numpyflat` (2D numpy array) mtype for `Panel` data... because @TonyBagnall insists on it 😄 

More specifically, this adds:
* checking routines for `numpyflat`
* conversion between `numpyflat` and any other `Panel` mtype currently supported
* fixture examples for the `numpyflat` mtype

Existing tests automatically cover both checks and conversions, nothing needed to be added there.

Also includes some convenience functionality:
* a new function `datatypes.scitype` which allows inference of the scitype, and tests for it
* new argument `exclude_mtypes` for various checker functions, since there are now ambiguous mtypes: numpy 2D can be a univariate `Panel`, or a multivariate `Series`, so by default `numpyflat` will be ignored in mtype inference, scitype inference, and scitype checks in favour of `numpy2D`.

### CODE REVIEW

File: sktime/datatypes/_panel/_check.py [hunk 1]
```diff
@@ -70,30 +77,25 @@ def _list_all_equal(obj):
 
 
 def check_dflist_Panel(obj, return_metadata=False, var_name="obj"):
```
Comment (chrisholder) [inline sktime/datatypes/_panel/_check.py:79]: Rename to check_dflist_panel

File: sktime/datatypes/_panel/_check.py [hunk 2]
```diff
@@ -70,30 +77,25 @@ def _list_all_equal(obj):
...
     n = len(obj)
 
     bad_inds = [i for i in range(n) if not isinstance(obj[i], pd.DataFrame)]
 
     if len(bad_inds) > 0:
         msg = f"{var_name}[i] must pd.DataFrame, but found other types at i={bad_inds}"
-        return ret(False, msg, None, return_metadata)
+        return _ret(False, msg, None, return_metadata)
 
     check_res = [check_pdDataFrame_Series(s, return_metadata=True) for s in obj]
```
Comment (chrisholder) [inline sktime/datatypes/_panel/_check.py:93]: Rename check_pdDataFrame_Series to check_pd_dataframe_series

File: sktime/datatypes/_panel/_check.py [hunk 3]
```diff
@@ -106,26 +108,21 @@ def ret(valid, msg, metadata, return_metadata):
...
     metadata["n_instances"] = n
 
-    return ret(True, None, metadata, return_metadata)
+    return _ret(True, None, metadata, return_metadata)
 
 
 check_dict[("df-list", "Panel")] = check_dflist_Panel
 
 
 def check_numpy3D_Panel(obj, return_metadata=False, var_name="obj"):
```
Comment (chrisholder) [inline sktime/datatypes/_panel/_check.py:117]: rename check_numpy3d_panel

File: sktime/datatypes/_panel/_check.py [hunk 4]
```diff
@@ -334,7 +319,37 @@ def ret(valid, msg, metadata, return_metadata):
...
     # end hacks
 
-    return ret(True, None, metadata, return_metadata)
+    return _ret(True, None, metadata, return_metadata)
 
 
 check_dict[("nested_univ", "Panel")] = is_nested_dataframe
+
+
+def check_numpyflat_Panel(obj, return_metadata=False, var_name="obj"):
```
Comment (chrisholder) [inline sktime/datatypes/_panel/_check.py:328]: Rename check_numpyflat_panel

File: sktime/datatypes/_panel/_convert.py [hunk 5]
```diff
@@ -1069,3 +1069,72 @@ def from_df_list_to_nested_adp(obj, store=None):
 
 
 convert_dict[("df-list", "nested_univ", "Panel")] = from_df_list_to_nested_adp
+
+
+def from_numpy3D_to_numpyflat(obj, store=None):
```
Comment (chrisholder) [inline sktime/datatypes/_panel/_convert.py:1074]: Rename from_numpy3d_to_numpyflat

File: sktime/datatypes/_panel/_convert.py [hunk 6]
```diff
@@ -1069,3 +1069,72 @@ def from_df_list_to_nested_adp(obj, store=None):
...
+
+    obj_in_2D = obj.reshape(shape[0], shape[1] * shape[2])
+
+    return obj_in_2D
+
+
+convert_dict[("numpy3D", "numpyflat", "Panel")] = from_numpy3D_to_numpyflat
+
+
+def from_numpyflat_to_numpy3D(obj, store=None):
```
Comment (chrisholder) [inline sktime/datatypes/_panel/_convert.py:1093]: rename from_numpyflat_to_numpy3d

**Review by chrisholder — CHANGES_REQUESTED** [review]

**Comment by fkiraly** [comment]
@chrisholder, your comments should be addressed.

**Pull request merged**